<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Adedotun Arogundade" />

<meta name="date" content="2025-07-21" />

<title>Exploring sagebrush microbial metagenomes from deep, host-derived sequencing</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="site_libs/viz-1.8.2/viz.js"></script>
<link href="site_libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="site_libs/grViz-binding-1.0.10/grViz.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Microbial Metagenomes</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Reprodicible Workflow</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Exploring sagebrush microbial metagenomes from deep, host-derived sequencing</h1>
<h3 class="subtitle">Reproducible Workflow</h3>
<h4 class="author">Adedotun Arogundade</h4>
<h4 class="date">2025-07-21</h4>

</div>


<!-- LOAD PACKAGES -->
<!-- SET GLOBAL OPTIONS -->
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The purpose of this project is to develop a reproducible workflow that is specific to my research entitled <strong>“Exploring sagebrush microbial metagenomes from deep, host-derived sequencing”</strong>.</p>
<p>The objectives of this study are to profile the microbial taxa associated with big sagebrush leaves, reconstruct MAGs where possible, and explore the potential functions of the resulting MAGs. We ask the following questions:</p>
<ol style="list-style-type: decimal">
<li><p>Which microbial taxa can we profile from deeply sequenced sagebrush shotgun reads?</p></li>
<li><p>How does host genotype affect microbial composition and diversity?</p></li>
<li><p>Are there differences in the community composition between greenhouse-grown vs. wild plants?</p></li>
</ol>
<div id="why-this-study" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Why this study?</h2>
<p>The traditional method for characterizing microbes is the isolation of individual strains from laboratory cultures <span class="citation">(<a href="#ref-Gulati_2020" role="doc-biblioref">Gulati and Plosky, 2020</a>; <a href="#ref-Anderson_and_Schluter_2021" role="doc-biblioref">Andersen and Schluter, 2021</a>)</span>. However, many microbes are difficult to culture because their natural habitat is too complex to be reproduced in the laboratory or because they rely on other species to successfully grow. This has limited the scope of microbial studies and has left many microbes uncharacterized <span class="citation">(<a href="#ref-Bharti2019" role="doc-biblioref">Bharti and Grimm, 2019</a>; <a href="#ref-Berg2020" role="doc-biblioref">Berg et al., 2020</a>)</span>. However, metagenomics sequencing data provides a culture-independent approach to studying complex communities of microbes <span class="citation">(<a href="#ref-Browne2016" role="doc-biblioref">Browne et al., 2016</a>; <a href="#ref-Nayfach2019" role="doc-biblioref">Nayfach et al., 2019</a>)</span>. This culture-independent approach includes taxonomic and functional profiling of microbes, as well as reconstruction of metagenome-assembled genomes (MAGs). A MAG refers to a collection of similar scaffolds that are grouped together from a metagenome assembly based on tetranucleotide frequencies (TNFs), abundances, complimentary marker genes <span class="citation">(<a href="#ref-Lin2016" role="doc-biblioref">Lin and Liao, 2016</a>)</span>, taxonomic alignments <span class="citation">(<a href="#ref-Wang2019" role="doc-biblioref">Wang et al., 2019</a>)</span> and codon usage <span class="citation">(<a href="#ref-Yu2018" role="doc-biblioref">Yu et al., 2018</a>)</span> which represents a microbial genome.</p>
</div>
<div id="why-sagebrush" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Why Sagebrush?</h2>
<p>Big sagebrush (Artemisia tridentata) is an important shrub species that dominates much of the western United States’ intermountain basins, shrub steppes, and deserts (USDA). The wide ecological attributes of sagebrush make it a foundational species affecting community composition, ecosystem processes, and wildlife habitat over vast portions of western North America <span class="citation">(<a href="#ref-Remington2021" role="doc-biblioref">Remington et al., 2021</a>)</span>. Its evergreen leaves offer nutrients and shade that facilitate the establishment of diverse understory plants even in arid environments. Sagebrush foliage is relatively unpalatable to livestock and other large herbivores due to high levels of terpenes and tannins, thereby buffering more palatable understory plants from intensive grazing <span class="citation">(<a href="#ref-Remington2021" role="doc-biblioref">Remington et al., 2021</a>)</span>. These grazing refuges help maintain native herbaceous diversity and allow sensitive species like perennial bunchgrasses to persist. Greater sage-grouse, pygmy rabbits, pronghorn, and mule deer rely on sagebrush for shelter and their primary winter food source <span class="citation">(<a href="#ref-Shipley2006" role="doc-biblioref">Shipley et al., 2006</a>)</span>. Certain songbirds, such as Brewer’s sparrow and sage thrasher, require intact sagebrush stands for nesting and rearing offspring. Sagebrush flowers are a vital nectar source for native pollinators including bees, wasps, flies, butterflies, and hummingbirds. Unfortunately, sagebrush ecosystems are critically threatened by wildfires, invasive species, climate change, land conversion, and urban development <span class="citation">(<a href="#ref-Miller2011" role="doc-biblioref">Miller, 2010</a>)</span> which has resulted in the loss of more than 50% of the historic sagebrush range <span class="citation">(<a href="#ref-Rigge2020" role="doc-biblioref">Rigge et al., 2020</a>)</span>. The major ecological role of sagebrush underscores the need to protect the remaining intact sagebrush communities through ecological restoration and sustainable management in order to conserve biodiversity and ecosystem services across the western U.S. <span class="citation">(<a href="#ref-Remington2021" role="doc-biblioref">Remington et al., 2021</a>)</span>.</p>
</div>
</div>
<div id="sampling" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Sampling</h1>
<!---
## Sample Distribution

The map below shows the source location of the sagebrush that were sequenced for this project.

-->
<div id="genomic-data" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Genomic Data</h2>
<p>We used sequence data from previous studies on sagebrush at Boise State University <span class="citation">(<a href="#ref-Melton2022" role="doc-biblioref">Melton et al., 2022</a>)</span>. These data are divided into three categories based on the environment in which the plants were grown prior to sequencing.</p>
<table>
<colgroup>
<col width="30%" />
<col width="25%" />
<col width="22%" />
<col width="21%" />
</colgroup>
<thead>
<tr class="header">
<th>Category</th>
<th>Sequencing Type</th>
<th>Sequencing Depth</th>
<th>Number of Sample</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Magenta box</td>
<td>Illumina paired-end</td>
<td>~160X</td>
<td>2</td>
</tr>
<tr class="even">
<td>Greenhouse</td>
<td>Illumina paired-end</td>
<td>~20X</td>
<td>12</td>
</tr>
<tr class="odd">
<td>Field/Wild</td>
<td>Illumina paired-end</td>
<td>~10-12X</td>
<td>14</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="bioinformatic-workflow" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Bioinformatic Workflow</h1>
<p>The workflow described here is specific to only one category of the samples (Field/Wild) that were analyzed for this project. However, the same workflow was used for all the other categories. As shown in Figure <a href="#fig:Pipeline">3.1</a>, the bioinformatic workflow can be described in five sections</p>
<ol style="list-style-type: decimal">
<li><p><a href="#Preprocessing"><strong>Preprocessing</strong></a></p></li>
<li><p><a href="#Taxonomic_assignment_of_short_reads"><strong>Taxonomic assignment of short reads</strong></a></p></li>
<li><p><a href="#AssemblyAndBinning"><strong>Assembly, and binning</strong></a></p></li>
<li><p><a href="#MAG_Classification"><strong>Classification of Metagenome Assembled Genomes</strong></a></p></li>
<li><p><a href="#FunctionalAnnotation"><strong>Gene Prediction and Functional profiling</strong></a></p></li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Pipeline"></span>
<img src="Images/Current_Working_Pipeline.png" alt="Overview of the bioinformatic workflow for the recovery of metagenome-assembled genomes from host-derived sequences of big sagebrush Artemisia tridentata subp. tridentata" width="100%" height="100%" />
<p class="caption">
Figure 3.1: Overview of the bioinformatic workflow for the recovery of metagenome-assembled genomes from host-derived sequences of big sagebrush Artemisia tridentata subp. tridentata
</p>
</div>
<div id="data-structure-and-organisation-of-files" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Data structure and organisation of files</h2>
<p>An overview of the project structure can be seen in Figure <a href="#fig:ProjectStructure">3.2</a>. The data visualization in Figure <a href="#fig:DataTree">3.3</a> enhances the understanding of the data flow and code.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ProjectStructure"></span>
<img src="Images/Project_Structure.png" alt="Overview of the project structure" width="100%" height="100%" />
<p class="caption">
Figure 3.2: Overview of the project structure
</p>
</div>
<ul>
<li><p>01_Bowtie : This contains the .bam file that was generated from previous preliminary mapping with host genome using bowtie2.</p></li>
<li><p>02_BMTAGER: The folder that contains the output of further clean up with BMTagger</p></li>
<li><p>03_KBASE_Analysis: The folder that contains the outputs from Kbase analysis platform</p></li>
<li><p>04_Kraken: Working folder for Kraken analysis.</p></li>
<li><p>05_DIAMOND: Working folder for DIAMOND analysis</p></li>
<li><p>06_Func_annotation: Working folder for the functional analysis</p></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:DataTree"></span>
<div class="grViz html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-1c86703c66b3c2eedef3" style="width:100%;height:100%;"></div>
<script type="application/json" data-for="htmlwidget-1c86703c66b3c2eedef3">{"x":{"diagram":"digraph {\n\ngraph [rankdir = \"LR\"]\n\n\n\n  \"1\" [label = \"Project_folder\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"2\" [label = \"01_Bowtie\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"3\" [label = \"Non_host_reads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"4\" [label = \"Clean-up\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"5\" [label = \"01_index.db\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"6\" [label = \"02_BMTAGGER\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"7\" [label = \"01_RAW_READS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"8\" [label = \"02_READ_QC\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"9\" [label = \"02_READ_QC-A_annua\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"10\" [label = \"02_READ_QC-human\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"11\" [label = \"03_CLEAN_READS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"12\" [label = \"03_CLEAN_READS-A_annua\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"13\" [label = \"03_CLEAN_READS-human\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"14\" [label = \"03_KBASE_Analysis\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"15\" [label = \"00_Host-removed_PE_reads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"16\" [label = \"01_Quality_Assessment\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"17\" [label = \"02_Read_Classification\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"18\" [label = \"kaiju_classifications\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"19\" [label = \"kaiju_summaries\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"20\" [label = \"krona_data\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"21\" [label = \"03_Assembly\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"22\" [label = \"12_GH_IDBA_Co-assembly.contigs.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"23\" [label = \"12_GH_MEGAHITmetalarge_Co-assembly.contigs.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"24\" [label = \"12_GH_MEGAHITmetasensitive_Co-assembly.contigs.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"25\" [label = \"12_GH_metaSPAdes_Co-assembly.contigs.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"26\" [label = \"04_Compare_Contigs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"27\" [label = \"histogram_figures\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"28\" [label = \"05_Binning\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"29\" [label = \"12_GH_metaSPAdes_Co-assembly_CONCOCT_BinnedContigs.FASTA_SET\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"30\" [label = \"12_GH_metaSPAdes_Co-assembly_MaxBin2_BinnedContigs.FASTA_SET\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"31\" [label = \"12_GH_metaSPAdes_Co-assembly_MetaBAT2_BinnedContigs.FASTA_SET\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"32\" [label = \"concoct_result\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"33\" [label = \"final_bins\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"34\" [label = \"maxbin_result\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"35\" [label = \"metabat_result\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"36\" [label = \"Bins\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"37\" [label = \"06_DASTool_Bin_Optimization\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"38\" [label = \"12_GH_metaSPAdes_Co-assembly_MaxBin2_MetaBAT_CONCOCT_DASTool.BinnedContigs.FASTA_SET\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"39\" [label = \"das_tool_result\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"40\" [label = \"das_tool_output_dir_DASTool_bins\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"41\" [label = \"07_CheckM_QualityAssessment\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"42\" [label = \"full_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"43\" [label = \"bins\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"44\" [label = \"bin.001\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"45\" [label = \"bin.002\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"46\" [label = \"bin.003\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"47\" [label = \"storage\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"48\" [label = \"aai_qa\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"49\" [label = \"bin.001\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"50\" [label = \"bin.002\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"51\" [label = \"bin.003\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"52\" [label = \"tree\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"53\" [label = \"tab_text\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"54\" [label = \"plots\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"55\" [label = \"08_Extract_Bin_Assemblies_From_DASToolBinnedContigs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"56\" [label = \"bin.001.fasta_assembly.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"57\" [label = \"bin.002.fasta_assembly.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"58\" [label = \"bin.003.fasta_assembly.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"59\" [label = \"HighQualityBins_50_10\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"60\" [label = \"Bin.001.fasta_assembly.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"61\" [label = \"Bin.002.fasta_assembly.FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"62\" [label = \"09_RAST_Annotation_of_Bins\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"63\" [label = \"bin.001.fasta_assembly.RAST.GFF___FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"64\" [label = \"bin.002.fasta_assembly.RAST.GFF___FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"65\" [label = \"bin.003.fasta_assembly.RAST.GFF___FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"66\" [label = \"GenBank_File\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"67\" [label = \"bin.001.fasta_assembly.RAST.GENBANK\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"68\" [label = \"bin.002.fasta_assembly.RAST.GENBANK\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"69\" [label = \"bin.003.fasta_assembly.RAST.GENBANK\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"70\" [label = \"HighQualityBins_50_10\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"71\" [label = \"Bin.001.fasta_assembly.RAST.GFF___FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"72\" [label = \"Bin.002.fasta_assembly.RAST.GFF___FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"73\" [label = \"10_GTDB_MAG_Classification\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"74\" [label = \"HighQualityBins_50_10\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"75\" [label = \"11_SpeciesTree-Phylogenetic-Placer\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"76\" [label = \"HighQualityBins_50_10\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"77\" [label = \"12_DRAM_Functional_Annotation\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"78\" [label = \"HighQualityBins_50_10\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"79\" [label = \"kb\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"80\" [label = \"module\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"81\" [label = \"work\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"82\" [label = \"tmp\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"83\" [label = \"DRAM_annos\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"84\" [label = \"genbank\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"85\" [label = \"13_dbCAN_Functional_FeatureSet\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"86\" [label = \"HMMER_dbCAN_Search.TAB\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"87\" [label = \"04_Kraken\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"88\" [label = \"00_database\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"89\" [label = \"nt_db\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"90\" [label = \"01_Host-removed_PE_reads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"91\" [label = \"02_PE_read_results\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"92\" [label = \"03_PE_read_results-nt_db\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"93\" [label = \"05_DIAMOND\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"94\" [label = \"01_Alignment-Host_removed_PE_reads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"95\" [label = \"06_Func_annotation\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"96\" [label = \"01_bins.fasta\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"97\" [label = \"02_KOFAM_Scan\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"98\" [label = \"01_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"99\" [label = \"01_KEGG\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"100\" [label = \"kofam_scan-1.3.0\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"101\" [label = \"lib\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"102\" [label = \"kofam_scan\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"103\" [label = \"cli\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"104\" [label = \"output_formatter\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"105\" [label = \"result\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"106\" [label = \"with_evalue_threshold\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"107\" [label = \"with_threshold_scale\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"108\" [label = \"with_threshold_scale_and_evalue_threshold\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"109\" [label = \"profiles\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"110\" [label = \"03_KEGG_Decoder\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"111\" [label = \"01_modified_FASTA\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"112\" [label = \"01_KOFAM_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"113\" [label = \"tmp\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"114\" [label = \"tabular\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"115\" [label = \"02_KEGG_decoder_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"116\" [label = \"03_KEGG_Expander\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"117\" [label = \"01_HMMModels\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"118\" [label = \"01_Pfam\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"119\" [label = \"02_NCBI_PGAP\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"120\" [label = \"BioData\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"121\" [label = \".git\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"122\" [label = \"branches\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"123\" [label = \"hooks\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"124\" [label = \"info\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"125\" [label = \"logs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"126\" [label = \"refs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"127\" [label = \"heads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"128\" [label = \"remotes\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"129\" [label = \"origin\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"130\" [label = \"objects\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"131\" [label = \"info\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"132\" [label = \"pack\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"133\" [label = \"refs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"134\" [label = \"heads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"135\" [label = \"remotes\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"136\" [label = \"origin\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"137\" [label = \"tags\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"138\" [label = \"KEGGDecoder\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"139\" [label = \"Amphibactin\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"140\" [label = \"DesferrioxamineBiosynthesis\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"141\" [label = \"DMSPLyase\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"142\" [label = \"DMSPSynthase\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"143\" [label = \"HMM_Models\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"144\" [label = \"PreviousExpanderDB_versions\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"145\" [label = \"images\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"146\" [label = \"MethylphosphonateSynthase\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"147\" [label = \"PreviousKEGGDecoderVersions\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"148\" [label = \"tests\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"149\" [label = \"tmp\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"150\" [label = \"tabular\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"151\" [label = \"04_MinPath\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"152\" [label = \"01_input\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"153\" [label = \"02_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"154\" [label = \"01_modified_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"155\" [label = \"MinPath\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"156\" [label = \".git\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"157\" [label = \"branches\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"158\" [label = \"hooks\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"159\" [label = \"info\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"160\" [label = \"logs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"161\" [label = \"refs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"162\" [label = \"heads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"163\" [label = \"remotes\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"164\" [label = \"origin\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"165\" [label = \"objects\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"166\" [label = \"info\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"167\" [label = \"pack\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"168\" [label = \"refs\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"169\" [label = \"heads\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"170\" [label = \"remotes\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"171\" [label = \"origin\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"172\" [label = \"tags\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"173\" [label = \"data\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"174\" [label = \"examples\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"175\" [label = \"glpk-4.6\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"176\" [label = \"autom4te.cache\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"177\" [label = \"contrib\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"178\" [label = \"deli\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"179\" [label = \"glpkmex\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"180\" [label = \"CVS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"181\" [label = \"doc\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"182\" [label = \"CVS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"183\" [label = \"examples\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"184\" [label = \"CVS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"185\" [label = \"src\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"186\" [label = \"CVS\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"187\" [label = \"jni\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"188\" [label = \"c\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"189\" [label = \"java\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"190\" [label = \"sample\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"191\" [label = \"doc\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"192\" [label = \"examples\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"193\" [label = \".deps\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"194\" [label = \"include\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"195\" [label = \"src\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"196\" [label = \".deps\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"197\" [label = \"sysdep\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"198\" [label = \"gnu\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"199\" [label = \"w32\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"200\" [label = \"Bin.001_Sphingomonas_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"201\" [label = \"Bin.002_Phyllobacterium_output\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"202\" [label = \"tmp\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n  \"203\" [label = \"tabular\", style = \"rounded\", shape = \"box\", fillcolor = \"#FFFFFF\", fontcolor = \"#000000\"] \n\"1\"->\"2\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"1\"->\"6\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"1\"->\"14\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"1\"->\"87\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"1\"->\"93\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"1\"->\"95\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"2\"->\"3\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"7\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"8\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"9\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"10\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"11\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"12\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"6\"->\"13\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"15\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"16\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"17\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"21\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"26\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"28\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"37\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"41\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"55\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"62\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"73\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"75\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"77\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"14\"->\"85\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"87\"->\"88\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"87\"->\"90\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"87\"->\"91\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"87\"->\"92\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"93\"->\"94\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"96\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"97\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"110\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"151\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"200\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"201\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"95\"->\"202\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"3\"->\"4\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"17\"->\"18\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"17\"->\"19\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"17\"->\"20\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"21\"->\"22\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"21\"->\"23\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"21\"->\"24\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"21\"->\"25\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"26\"->\"27\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"29\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"30\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"31\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"32\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"34\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"28\"->\"35\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"37\"->\"38\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"37\"->\"39\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"41\"->\"42\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"41\"->\"54\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"55\"->\"56\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"55\"->\"57\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"55\"->\"58\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"55\"->\"59\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"62\"->\"63\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"62\"->\"64\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"62\"->\"65\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"62\"->\"66\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"62\"->\"70\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"73\"->\"74\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"75\"->\"76\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"77\"->\"78\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"77\"->\"79\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"85\"->\"86\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"88\"->\"89\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"97\"->\"98\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"97\"->\"100\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"97\"->\"109\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"110\"->\"111\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"151\"->\"152\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"151\"->\"153\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"151\"->\"155\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"202\"->\"203\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"4\"->\"5\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"32\"->\"33\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"35\"->\"36\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"39\"->\"40\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"42\"->\"43\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"42\"->\"47\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"42\"->\"53\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"59\"->\"60\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"59\"->\"61\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"66\"->\"67\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"66\"->\"68\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"66\"->\"69\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"70\"->\"71\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"70\"->\"72\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"79\"->\"80\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"98\"->\"99\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"100\"->\"101\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"111\"->\"112\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"111\"->\"115\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"111\"->\"116\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"111\"->\"149\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"153\"->\"154\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"155\"->\"156\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"155\"->\"173\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"155\"->\"174\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"155\"->\"175\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"43\"->\"44\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"43\"->\"45\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"43\"->\"46\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"47\"->\"48\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"47\"->\"52\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"80\"->\"81\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"101\"->\"102\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"112\"->\"113\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"116\"->\"117\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"116\"->\"120\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"149\"->\"150\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"157\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"158\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"159\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"160\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"165\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"156\"->\"168\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"176\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"177\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"191\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"192\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"194\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"195\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"175\"->\"197\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"48\"->\"49\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"48\"->\"50\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"48\"->\"51\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"81\"->\"82\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"102\"->\"103\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"102\"->\"104\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"102\"->\"105\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"113\"->\"114\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"117\"->\"118\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"117\"->\"119\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"120\"->\"121\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"120\"->\"138\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"120\"->\"148\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"160\"->\"161\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"165\"->\"166\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"165\"->\"167\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"168\"->\"169\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"168\"->\"170\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"168\"->\"172\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"177\"->\"178\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"177\"->\"179\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"177\"->\"187\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"192\"->\"193\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"195\"->\"196\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"197\"->\"198\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"197\"->\"199\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"82\"->\"83\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"105\"->\"106\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"105\"->\"107\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"105\"->\"108\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"122\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"123\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"124\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"125\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"130\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"121\"->\"133\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"139\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"140\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"141\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"142\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"143\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"145\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"146\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"138\"->\"147\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"161\"->\"162\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"161\"->\"163\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"170\"->\"171\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"179\"->\"180\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"179\"->\"181\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"179\"->\"183\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"179\"->\"185\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"187\"->\"188\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"187\"->\"189\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"187\"->\"190\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"83\"->\"84\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"125\"->\"126\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"130\"->\"131\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"130\"->\"132\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"133\"->\"134\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"133\"->\"135\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"133\"->\"137\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"143\"->\"144\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"163\"->\"164\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"181\"->\"182\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"183\"->\"184\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"185\"->\"186\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"126\"->\"127\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"126\"->\"128\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"135\"->\"136\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n\"128\"->\"129\" [arrowhead = \"vee\", color = \"grey\", penwidth = \"2px\"] \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 3.3: Data/Project Visualization
</p>
</div>
</div>
<div id="bioinformatic-tool" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Bioinformatic Tool</h2>
<p>Analyses presented in this report requires a UNIX-based Operating System and <code>ssh</code>, <code>sbatch</code> and <code>tmux</code> protocols should be installed. Check this <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Tutorials.html">website</a> for more details and documentation on how to connect to remote computers using the <code>Secure Shell Protocol</code>.</p>
<p>In addition, R and Rstudio was used to tidy/format the output of data and to produce figures. All the R analyses were performed using R version 4.2.1 (2022-06-23). The software can be downloaded <a href="https://posit.co/download/rstudio-desktop/">here</a>.</p>
<p>KBase (Web-based analyses) was accessed using Google Chrome Version 119.0.6045.105 (Official Build) (x86_64). You can install Google Chrome using this <a href="https://www.google.com/chrome/">link</a></p>
<p>Finally, dependencies associated to each analysis will be presented alongside the code.</p>
</div>
<div id="setting-up" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Setting Up!</h2>
<pre class="r"><code># 1. Remotely connect to HPC computer using ssh (adjust with your credentials/IP)

ssh userID@IP

# 2. Start a dev-session 

dev-session bsu

# 3. Start tmux session and name it Sagebrush_MAGS. 

tmux new -s Sagebrush_MAGS

 # Note: Use Ctrl+b and d to safe exit a tmux session

# 3.  Navigate to project location (adjust path)

cd /path/to/Project_folder/</code></pre>
<div id="InstallMetagenomicsEnv" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Create a metagenomics conda environment</h3>
<p>Create a conda environment from <a href="https://github.com/carpentries-lab/metagenomics-analysis">Data carpentry tutorial</a>, and name the environment <strong>“metagenomics”</strong>. This environment will contain several software that will be used in this project.</p>
<pre class="r"><code>wget https://raw.githubusercontent.com/carpentries-lab/metagenomics-analysis/gh-pages/files/spec-file-Ubuntu22.txt

#Then run to create an environment called &#39;metagenomics&#39; using the yml.txt file that you donwloaded (spec-file-Ubuntu22.txt)

conda create --name metagenomics --file spec-file-Ubuntu22.txt

# Activate the metagenomics environment

conda activate metagenomics</code></pre>
</div>
</div>
<div id="Preprocessing" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Preprocessing</h2>
<p>The sequences data in this study are host-derived as they were extracted from plant tissues, and with our focus on the associated microbial reads, we consider host genomes as contaminants that could impact our analyses. Therefore the need to remove them. Raw sequence data from previous projects had been initially mapped with sagebrush (Artemisia tridentata subp. tridentata) genome to generate <strong>.bam</strong> file which are stored in <strong>01_Bowtie folder</strong>. See Figure <a href="#fig:Pipeline">3.1</a>.</p>
<div id="seperate-unmapped-reads-from-mapped-reads" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Seperate <em>‘unmapped’</em> reads from <em>‘mapped’</em> reads</h3>
<div id="data-requirement" class="section level4" number="3.4.1.1">
<h4><span class="header-section-number">3.4.1.1</span> Data requirement</h4>
<p><strong>- Input:</strong> <code>bam</code> files from Bowtie2</p>
<p><strong>-Output:</strong> Unmapped non-host <code>fastq.gz</code> files.</p>
</div>
<div id="dependencies" class="section level4" number="3.4.1.2">
<h4><span class="header-section-number">3.4.1.2</span> Dependencies</h4>
<p>This is done using <code>samtools</code> which can be installed using Bioconda <a href="https://anaconda.org/bioconda/samtools">here</a></p>
<pre class="r"><code>conda install -c bioconda samtools</code></pre>
</div>
<div id="code" class="section level4" number="3.4.1.3">
<h4><span class="header-section-number">3.4.1.3</span> Code</h4>
<p>Here, we used a ’for loop` code that contain different code that did all the process described below.</p>
<ol style="list-style-type: decimal">
<li><p>We removed the host reads from the bam files</p></li>
<li><p>Then retrieved the unmapped non-host reads</p></li>
<li><p>Made it a sorted bam</p></li>
<li><p>Split the unmapped_sorted.bam file into R1 and R2</p></li>
<li><p>Finally, I deleted the unwanted intermediate “unmapped.bam” and “unmapped_sorted.bam” files.</p></li>
</ol>
<pre class="r"><code># Step 1: Remove the host reads from the bam files

for F in /path/to/Project_folder/01_Bowtie/*.bam; do
    PWD=/path/to/Project_folder/
    BASENAME=${F##*/}
    SAMPLE=${BASENAME%.bam}_IDT3_NC
    OUTPUT_DIR=$PWD/01_Bowtie/Non_host_reads


# Step 2: Get the unmapped non-host reads

samtools view -b -f 12 -F 256 $F  &gt; $OUTPUT_DIR/${SAMPLE}_unmapped.bam &amp;&amp;

# Step 3: Make it a sorted bam

samtools sort -n -m 5G -@ 2 $OUTPUT_DIR/${SAMPLE}_unmapped.bam -o $OUTPUT_DIR/${SAMPLE}_unmapped_sorted.bam &amp;&amp;

# Step 4: split the unmapped_sorted.bam file into R1 and R2

samtools fastq -@ 8 $OUTPUT_DIR/${SAMPLE}_unmapped_sorted.bam \
  -1 $OUTPUT_DIR/${SAMPLE}_host_removed_R1.fastq.gz \
  -2 $OUTPUT_DIR/${SAMPLE}_host_removed_R2.fastq.gz \
  -0 /dev/null -s /dev/null -n &amp;&amp;

# Step 5: Remove the intermediate &quot;unmapped.bam&quot; and &quot;unmapped_sorted.bam&quot; files

rm $OUTPUT_DIR/${SAMPLE}_unmapped.bam &amp;&amp;
rm $OUTPUT_DIR/${SAMPLE}_unmapped_sorted.bam

done</code></pre>
</div>
</div>
<div id="removal-of-host-reads-using-bmtagger" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Removal of ‘host’ reads using BMTagger</h3>
<p>Downstream analysis with the “unmapped non-host” reads from Bowtie2 showed that there were still sagebrush reads in our “non-host” reads therefore, we used BMTagger, a tool specifically designed to remove the human genome from metagenomic datasets but can be adapted to remove other host genomes. See Figure <a href="#fig:Pipeline">3.1</a>.</p>
<p>In order to improve the clean-up process, we indexed reference genomes from <em>Artemisia tridentata</em>, <em>Artemisia annua</em>, and the <em>human hg38 genome</em> which were downloaded on NCBI and mapped them with our datasets.</p>
<div id="dependencies-1" class="section level4" number="3.4.2.1">
<h4><span class="header-section-number">3.4.2.1</span> Dependencies</h4>
<p>For this, we used the one of the modules of <strong>Metawrap</strong> called <code>read_qc.sh</code>. We prefered this module because it does the following all at once:</p>
<ol style="list-style-type: decimal">
<li><p>Quality Assessment using <code>fastqc</code>.</p></li>
<li><p>Trims bad sequence data using <code>trim_galore</code>.</p></li>
<li><p>Removes host reads using <code>bmtagger</code> by mapping against the host build <code>.bitmask</code> and <code>.srprism</code> files.</p></li>
</ol>
<p>Install metawrap and all of it dependencies using this code. Visit this <a href="https://github.com/bxlab/metaWRAP">github</a> for more information and on how to build the ‘host’ BMTagger database.</p>
<pre class="r"><code>conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels bioconda
conda config --add channels ursky

# Unix/Linux only
mamba create -y -n metawrap-env python=2.7

conda activate metawrap-env</code></pre>
</div>
<div id="create-the-bmtagger-folders" class="section level4" number="3.4.2.2">
<h4><span class="header-section-number">3.4.2.2</span> Create the BMTagger folders</h4>
<p>Set up for BMTagger clean up</p>
<pre class="r"><code># Create a BMTAGGER folder 

mkdir /path/to/Project_folder/02_BMTAGGER       # This is going to be my PWD

# Reference: https://github.com/bxlab/metaWRAP/blob/master/Usage_tutorial.md

# Change to the Bmtagger folder

cd /path/to/Project_folders/02_BMTAGGER

# Make a new folder called 01_RAW_READS

mkdir /path/to/Project_folder/02_BMTAGGER/01_RAW_READS

cd /path/to/Project_folder/02_BMTAGGER/01_RAW_READS

# Copy the raw non-host reads in &#39;/path/to/Project_folder/01_Bowtie/Non_host_reads&#39; into &#39;/path/to/Project_folder/02_BMTAGGER/01_RAW_READS&#39;

cp /path/to/Project_folder/01_Bowtie/Non_host_reads/* /path/to/Project_folder/02_BMTAGGER/01_RAW_READS

# Unzip the files in 01_RAW_READS (change the extension from .fastq.gz to .fastq)

cd /path/to/Project_folder/02_BMTAGGER/01_RAW_READS

gunzip *.gz</code></pre>
</div>
<div id="clean-with-artemisia-tridenta-subp-tridenta-genome" class="section level4" number="3.4.2.3">
<h4><span class="header-section-number">3.4.2.3</span> Clean with <em>Artemisia tridenta</em> subp <em>tridenta</em> genome</h4>
<p>See Figure <a href="#fig:Pipeline">3.1</a>.</p>
<pre class="r"><code># Create a new tmux session and change to the working directory

tmux new -s bmtagger

cd /path/to/Project_folder/02_BMTAGGER

# Activate the conda environment in this new tmux session

conda activate metawrap-env


# Run metaWRAP-Read_qc to trim the reads, remove host (A_tridentata) reads

# Make a directory for the results

mkdir /path/to/Project_folder/02_BMTAGGER/02_READ_QC


# Process all samples at the same time with a parallel for loop (especially if you have many samples)

# Note: my PWD is /path/to/Project_folder/02_BMTAGGER 

# Raw reads are found in /path/to/Project_folder/02_BMTAGGER/01_RAW_READS

for F in 01_RAW_READS/*_R1.fastq; do 
    R=${F%_*}_R2.fastq
    BASE=${F##*/}
    SAMPLE=${BASE%_host*}
    metawrap read_qc -1 $F -2 $R -t 20 -x A_tridentata -o 02_READ_QC/$SAMPLE --skip-trimming &amp;
done    


# change to /path/to/Project_folder/02_BMTAGGER
    
cd /path/to/Project_folder/02_BMTAGGER


# move over the final QC&#39;ed reads into a new folder but first make a new folder that will store the reads

mkdir /path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS


for i in 02_READ_QC/*; do 
 b=${i#*/}
 mv ${i}/final_pure_reads_1.fastq 03_CLEAN_READS/${b}_1.fastq
 mv ${i}/final_pure_reads_2.fastq 03_CLEAN_READS/${b}_2.fastq
done</code></pre>
</div>
<div id="clean-with-artemisia-annua-genome" class="section level4" number="3.4.2.4">
<h4><span class="header-section-number">3.4.2.4</span> Clean with <em>Artemisia annua</em> genome</h4>
<p>See Figure <a href="#fig:Pipeline">3.1</a>.</p>
<p>The A. annua genome that was used was downloaded from NCBI under <a href="https://www.ncbi.nlm.nih.gov/Traces/wgs/PKPP01?display=download">Bioproject PRJNA416223</a>.</p>
<p>The reads in <code>/path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS</code> have been cleaned of <em>A. tridentata</em> reads. Now, I want to clean them with <em>A. annua</em> reads for further clean-up due to the similarity of these genomes.</p>
<p>Now, these reads in <code>/path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS</code> will serve as template for this new cleaning process</p>
<pre class="r"><code># Make a folder that will store the output

mkdir /path/to/Project_folder/02_BMTAGGER/02_READ_QC-A_annua


for F in 03_CLEAN_READS/*_1.fastq; do 
 R=${F%_*}_2.fastq
 BASE=${F##*/}
 SAMPLE=${BASE%_*}
 metawrap read_qc -1 $F -2 $R -t 5 -x A_annua -o 02_READ_QC-A_annua/$SAMPLE --skip-trimming &amp;
done




## move over the final QC&#39;ed reads into a new folder

mkdir /path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS-A_annua


for i in 02_READ_QC-A_annua/*; do 
 b=${i#*/}
 mv ${i}/final_pure_reads_1.fastq 03_CLEAN_READS-A_annua/${b}_1.fastq
 mv ${i}/final_pure_reads_2.fastq 03_CLEAN_READS-A_annua/${b}_2.fastq
done</code></pre>
</div>
<div id="clean-with-human-homo-sapien-genome" class="section level4" number="3.4.2.5">
<h4><span class="header-section-number">3.4.2.5</span> Clean with human (<em>Homo sapien</em>) genome</h4>
<p>See Figure <a href="#fig:Pipeline">3.1</a>.</p>
<p>The reads in <code>/path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS-A_annuaS</code> have been cleaned of <em>A. tridentata</em> and <em>A. annua</em> reads. Now, I want to clean them with <em>Homo sapien</em> genome to remove any potential contaminants that may get carried forward from the DNA extraction process.</p>
<p>The human genome that was used was downloaded from <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/" class="uri">https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/</a> using <code>wget</code> and <code>ftp</code> protocol: <code>wget ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/*fa.gz</code>. Check this <a href="https://github.com/bxlab/metaWRAP/blob/master/installation/database_installation.md">link</a> for details.</p>
<p>Now, these reads in <code>/path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS-A_annua</code> will serve as template for this new cleaning process</p>
<pre class="r"><code># Make a folder that will store the output

mkdir /path/to/Project_folder/02_BMTAGGER/02_READ_QC-human


for F in 03_CLEAN_READS-A_annua/*_1.fastq; do 
 R=${F%_*}_2.fastq
 BASE=${F##*/}
 SAMPLE=${BASE%_*}
 metawrap read_qc -1 $F -2 $R -t 5 -x hg38 -o 02_READ_QC-human/$SAMPLE --skip-trimming &amp;
done


## move over the final QC&#39;ed reads into a new folder

mkdir /path/to/Project_folder/02_BMTAGGER/03_CLEAN_READS-human


for i in 02_READ_QC-human/*; do 
 b=${i#*/}_host_removed
 mv ${i}/final_pure_reads_1.fastq 03_CLEAN_READS-human/${b}_1.fastq
 mv ${i}/final_pure_reads_2.fastq 03_CLEAN_READS-human/${b}_2.fastq
done


# Note that I added the phrase &quot;host_removed&quot; to the final cleaned read</code></pre>
</div>
<div id="output" class="section level4" number="3.4.2.6">
<h4><span class="header-section-number">3.4.2.6</span> Output</h4>
<p>Table <a href="#tab:CleanReads">3.1</a> shows the result of the cleaning process</p>
<table class=" lightable-paper lightable-hover" style="font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:CleanReads">Table 3.1: </span>Approximate size of ‘host’ cleaned reads that were removed from the sequencing data.
</caption>
<thead>
<tr>
<th style="text-align:center;">
Sample_ID
</th>
<th style="text-align:center;">
Category
</th>
<th style="text-align:center;">
Initial_read_size
</th>
<th style="text-align:center;">
Approximate_Size_of_A.tridentata_cleaned_reads
</th>
<th style="text-align:center;">
Approximate_Size_of_A.annua_cleaned_reads
</th>
<th style="text-align:center;">
Approximate_Size_of_human_cleaned_reads
</th>
<th style="text-align:center;">
Percentage_Cleaned_Reads
</th>
<th style="text-align:center;">
Final_read_size
</th>
<th style="text-align:center;">
No_of_cleaned_PE_raw_reads_in_basepairs
</th>
</tr>
</thead>
<tbody>
<tr grouplength="2">
<td colspan="9" style="border-bottom: 1px solid;">
<strong>Magenta Samples</strong>
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
IDT3
</td>
<td style="text-align:center;">
Magenta
</td>
<td style="text-align:center;">
38.0 Gb
</td>
<td style="text-align:center;">
24.0 Gb
</td>
<td style="text-align:center;">
4.4 Gb
</td>
<td style="text-align:center;">
970 Mb
</td>
<td style="text-align:center;">
77.28947
</td>
<td style="text-align:center;">
8.4 Gb
</td>
<td style="text-align:center;">
25,738,784
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
UTT2
</td>
<td style="text-align:center;">
Magenta
</td>
<td style="text-align:center;">
58.0 Gb
</td>
<td style="text-align:center;">
34.0 Gb
</td>
<td style="text-align:center;">
4.6 Gb
</td>
<td style="text-align:center;">
1.31 Gb
</td>
<td style="text-align:center;">
68.81034
</td>
<td style="text-align:center;">
18.2 Gb
</td>
<td style="text-align:center;">
55,350,386
</td>
</tr>
<tr grouplength="12">
<td colspan="9" style="border-bottom: 1px solid;">
<strong>Greenhouse Samples</strong>
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
29-ID
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
4.4 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
658 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
64.95455
</td>
<td style="text-align:center;">
1.48 Gb
</td>
<td style="text-align:center;">
4,207,818
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
143-ID
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.6 Gb
</td>
<td style="text-align:center;">
1.83 Gb
</td>
<td style="text-align:center;">
550 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
66.11111
</td>
<td style="text-align:center;">
1.29 Gb
</td>
<td style="text-align:center;">
3,683,012
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
163-ID
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.4 Gb
</td>
<td style="text-align:center;">
1.84 Gb
</td>
<td style="text-align:center;">
540 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
70.00000
</td>
<td style="text-align:center;">
1.11 Gb
</td>
<td style="text-align:center;">
3,158,346
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
192-ID
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.4 Gb
</td>
<td style="text-align:center;">
1.66 Gb
</td>
<td style="text-align:center;">
456 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
62.23529
</td>
<td style="text-align:center;">
1.31 Gb
</td>
<td style="text-align:center;">
3,717,942
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
246-NV
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
4.2 Gb
</td>
<td style="text-align:center;">
2.20 Gb
</td>
<td style="text-align:center;">
602 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
66.71429
</td>
<td style="text-align:center;">
1.58 Gb
</td>
<td style="text-align:center;">
4,490,656
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
326-NV
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.6 Gb
</td>
<td style="text-align:center;">
1.98 Gb
</td>
<td style="text-align:center;">
576 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
71.00000
</td>
<td style="text-align:center;">
1.15 Gb
</td>
<td style="text-align:center;">
3,267,482
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
379-NV
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.8 Gb
</td>
<td style="text-align:center;">
2.02 Gb
</td>
<td style="text-align:center;">
560 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
67.89474
</td>
<td style="text-align:center;">
1.29 Gb
</td>
<td style="text-align:center;">
3,666,774
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
399-NV
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.40 Gb
</td>
<td style="text-align:center;">
580 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
74.50000
</td>
<td style="text-align:center;">
1.21 Gb
</td>
<td style="text-align:center;">
3,440,062
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
422-UT
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
5.8 Gb
</td>
<td style="text-align:center;">
3.20 Gb
</td>
<td style="text-align:center;">
780 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
68.62069
</td>
<td style="text-align:center;">
2.02 Gb
</td>
<td style="text-align:center;">
5,794,018
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
443-UT
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.02 Gb
</td>
<td style="text-align:center;">
562 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
64.55000
</td>
<td style="text-align:center;">
1.47 Gb
</td>
<td style="text-align:center;">
4,186,566
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
541-UT
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.20 Gb
</td>
<td style="text-align:center;">
620 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
70.50000
</td>
<td style="text-align:center;">
1.25 Gb
</td>
<td style="text-align:center;">
3,558,232
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
579-UT
</td>
<td style="text-align:center;">
Greenhouse
</td>
<td style="text-align:center;">
3.8 Gb
</td>
<td style="text-align:center;">
1.98 Gb
</td>
<td style="text-align:center;">
564 Mb
</td>
<td style="text-align:center;">
0 Mb
</td>
<td style="text-align:center;">
66.94737
</td>
<td style="text-align:center;">
1.30 Gb
</td>
<td style="text-align:center;">
3,696,426
</td>
</tr>
<tr grouplength="14">
<td colspan="9" style="border-bottom: 1px solid;">
<strong>Field/Wild Samples</strong>
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
2-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.2 Gb
</td>
<td style="text-align:center;">
1.8 Gb
</td>
<td style="text-align:center;">
464 Mb
</td>
<td style="text-align:center;">
4.4 Mb
</td>
<td style="text-align:center;">
70.88750
</td>
<td style="text-align:center;">
911 Mb
</td>
<td style="text-align:center;">
2,601,426
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
3-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.4 Gb
</td>
<td style="text-align:center;">
2.0 Gb
</td>
<td style="text-align:center;">
474 Mb
</td>
<td style="text-align:center;">
3.6 Mb
</td>
<td style="text-align:center;">
72.87059
</td>
<td style="text-align:center;">
907 Mb
</td>
<td style="text-align:center;">
2,590,152
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
8-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.0 Gb
</td>
<td style="text-align:center;">
524 Mb
</td>
<td style="text-align:center;">
2.6 Mb
</td>
<td style="text-align:center;">
63.16500
</td>
<td style="text-align:center;">
2.3 Gb
</td>
<td style="text-align:center;">
6,519,242
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
14-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.0 Gb
</td>
<td style="text-align:center;">
2.0 Gb
</td>
<td style="text-align:center;">
502 Mb
</td>
<td style="text-align:center;">
1.0 Mb
</td>
<td style="text-align:center;">
83.43333
</td>
<td style="text-align:center;">
853 Mb
</td>
<td style="text-align:center;">
2,436,802
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
15-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
588 Mb
</td>
<td style="text-align:center;">
48 Mb
</td>
<td style="text-align:center;">
70.90000
</td>
<td style="text-align:center;">
1.2 Gb
</td>
<td style="text-align:center;">
3,604,450
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
21-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
540 Mb
</td>
<td style="text-align:center;">
12.8 Mb
</td>
<td style="text-align:center;">
68.82000
</td>
<td style="text-align:center;">
1.3 Gb
</td>
<td style="text-align:center;">
3,559,682
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
24-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.0 Gb
</td>
<td style="text-align:center;">
1.7 Gb
</td>
<td style="text-align:center;">
470 Mb
</td>
<td style="text-align:center;">
1.9 Mb
</td>
<td style="text-align:center;">
72.39667
</td>
<td style="text-align:center;">
800 Mb
</td>
<td style="text-align:center;">
2,283,504
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
26-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.6 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
548 Mb
</td>
<td style="text-align:center;">
13.4 Mb
</td>
<td style="text-align:center;">
76.70556
</td>
<td style="text-align:center;">
1.0 Gb
</td>
<td style="text-align:center;">
2,960,114
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
29-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.6 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
448 Mb
</td>
<td style="text-align:center;">
2.2 Mb
</td>
<td style="text-align:center;">
73.61667
</td>
<td style="text-align:center;">
853 Mb
</td>
<td style="text-align:center;">
2,433,302
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
32-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.4 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
510 Mb
</td>
<td style="text-align:center;">
1.5 Mb
</td>
<td style="text-align:center;">
79.75000
</td>
<td style="text-align:center;">
911 Mb
</td>
<td style="text-align:center;">
2,590,152
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
33-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.8 Gb
</td>
<td style="text-align:center;">
2.2 Gb
</td>
<td style="text-align:center;">
512 Mb
</td>
<td style="text-align:center;">
3.6 Mb
</td>
<td style="text-align:center;">
71.46316
</td>
<td style="text-align:center;">
1.2 Gb
</td>
<td style="text-align:center;">
3,528,676
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
34-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.4 Gb
</td>
<td style="text-align:center;">
1.9 Gb
</td>
<td style="text-align:center;">
488 Mb
</td>
<td style="text-align:center;">
8.4 Mb
</td>
<td style="text-align:center;">
70.48235
</td>
<td style="text-align:center;">
1.1 Gb
</td>
<td style="text-align:center;">
3,125,510
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
35-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
4.0 Gb
</td>
<td style="text-align:center;">
1.8 Gb
</td>
<td style="text-align:center;">
234 Mb
</td>
<td style="text-align:center;">
556 Mb
</td>
<td style="text-align:center;">
64.75000
</td>
<td style="text-align:center;">
1.3 Gb
</td>
<td style="text-align:center;">
3,839,712
</td>
</tr>
<tr>
<td style="text-align:center;padding-left: 2em;" indentlevel="1">
36-Wild
</td>
<td style="text-align:center;">
Field/Wild
</td>
<td style="text-align:center;">
3.6 Gb
</td>
<td style="text-align:center;">
2.0 Gb
</td>
<td style="text-align:center;">
518 Mb
</td>
<td style="text-align:center;">
3.0 Mb
</td>
<td style="text-align:center;">
70.02778
</td>
<td style="text-align:center;">
965 Mb
</td>
<td style="text-align:center;">
2,754,048
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="Taxonomic_assignment_of_short_reads" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Taxonomic assignment of short reads</h2>
<p>As shown in Figure <a href="#fig:Pipeline">3.1</a>, the Department of Energy Systems Biology Knowledgebase (KBase) System was used for our downstream metagenomic analyses <span class="citation">(<a href="#ref-Arkin2018" role="doc-biblioref">Arkin et al., 2018</a>)</span> with default parameters for all software unless otherwise stated. You can sign up to Kbase using this <a href="https://www.kbase.us/">link</a>.</p>
<p>Figure <a href="#fig:KbaseOverview">3.4</a> below shows the Kbase pipeline that was used in this study. You can watch the short video below on how to get started on Kbase <a href="https://www.youtube.com/watch?v=WKKKXg55hs0">here</a></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseOverview"></span>
<img src="Images/Overview_of_Kbase.png" alt="Overview of Kbase Pipeline" width="100%" height="100%" />
<p class="caption">
Figure 3.4: Overview of Kbase Pipeline
</p>
</div>
<div id="import-into-kbase" class="section level3" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Import into Kbase</h3>
<p>Start by importing the sets of paired-end reads in FASTQ format as described in Figure <a href="#fig:KbaseImport">3.5</a>. The Import App creates a PairedEndLibrary object that we can then run through FastQC to further check the quality of the reads.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseImport"></span>
<img src="Images/Kbase_ImportData.png" alt="Screenshot on how to Import Data into Kbase" width="100%" height="100%" />
<p class="caption">
Figure 3.5: Screenshot on how to Import Data into Kbase
</p>
</div>
</div>
<div id="run-fastqc" class="section level3" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Run FastQC</h3>
<div id="data-requirement-1" class="section level4" number="3.5.2.1">
<h4><span class="header-section-number">3.5.2.1</span> Data Requirement</h4>
<ul>
<li><p><strong>Input</strong>: PairedEndLibrary that was generated using the Import Application</p></li>
<li><p><strong>Output</strong>: Quality Report in HTML format.</p></li>
</ul>
</div>
<div id="dependencies-2" class="section level4" number="3.5.2.2">
<h4><span class="header-section-number">3.5.2.2</span> Dependencies</h4>
<p>We used <code>Assess Read Quality with FastQC - v0.12.1</code> for further check the quality of our data. An example of the result can be is on Figure <a href="#fig:KbaseFastQC">3.6</a></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseFastQC"></span>
<img src="Images/KbaseFastQC.png" alt="Screenshot of a FastQC report for one of the samples" width="100%" height="100%" />
<p class="caption">
Figure 3.6: Screenshot of a FastQC report for one of the samples
</p>
</div>
</div>
</div>
<div id="taxonomic-assignment-of-short-reads-with-kaiju" class="section level3" number="3.5.3">
<h3><span class="header-section-number">3.5.3</span> Taxonomic Assignment of short reads with Kaiju</h3>
<div id="data-requirement-2" class="section level4" number="3.5.3.1">
<h4><span class="header-section-number">3.5.3.1</span> Data Requirement</h4>
<ul>
<li><p><strong>Input</strong>: PairedEndLibrary that was generated using the Import Application</p></li>
<li><p><strong>Output</strong>: Taxonomic visualization plots. It also produces a taxonomic report that can be exported into other programs to generate better quality plots.</p></li>
</ul>
</div>
<div id="dependencies-3" class="section level4" number="3.5.3.2">
<h4><span class="header-section-number">3.5.3.2</span> Dependencies</h4>
<p>We used <code>Classify Taxonomy of Metagenomic Reads with Kaiju - v1.9.0</code> on Kbase to classify the short Illumina reads. The software was configured as follows</p>
<pre class="r"><code>Taxonomic Level           = ALL

Reference DB              = NCBI BLAST nr + euk
  
Low Abundance Filter      = 0.5

Subsample percent         = 100</code></pre>
<p>The outputs of Kaiju were exported out of Kbase and used to plot bar graph. Figure <a href="#fig:KbaseFastQC">3.6</a> is a barplot generated using the outputs from KBase.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:TaxaPlot"></span>
<img src="Images/Taxonomic.Barplots.png" alt="Taxonomic barplot showing the relative abundance of the top taxa across A) Magenta samples B) Greenhouse samples C) Wild/Field Samples at the genus level." width="100%" height="100%" />
<p class="caption">
Figure 3.7: Taxonomic barplot showing the relative abundance of the top taxa across A) Magenta samples B) Greenhouse samples C) Wild/Field Samples at the genus level.
</p>
</div>
</div>
</div>
<div id="taxonomic-assignment-of-short-reads-with-kraken2-and-diamond" class="section level3" number="3.5.4">
<h3><span class="header-section-number">3.5.4</span> Taxonomic Assignment of short reads with Kraken2 and DIAMOND</h3>
<p>We also used other classification tool such as Kraken2 and DIAMOND/MEGAN <span class="citation">(<a href="#ref-Buchfink2015" role="doc-biblioref">Buchfink et al., 2015</a>; <a href="#ref-Wood2019" role="doc-biblioref">Wood et al., 2019</a>)</span> so that we can compare the results of Kaiju with other softwares. We found that the results are very similar. In order to keep this project concise, we will not explain the details of these software.</p>
</div>
</div>
<div id="AssemblyAndBinning" class="section level2" number="3.6">
<h2><span class="header-section-number">3.6</span> Assembly, and binning</h2>
<div id="assembly" class="section level3" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Assembly</h3>
<div id="data-requirement-3" class="section level4" number="3.6.1.1">
<h4><span class="header-section-number">3.6.1.1</span> Data Requirement</h4>
<ul>
<li><p><strong>Input</strong>: PairedEndLibrary that was generated using the Import Application or Merged Read library if you have multiple samples.</p></li>
<li><p><strong>Output</strong>: Output ContigSet.</p></li>
</ul>
</div>
<div id="dependencies-4" class="section level4" number="3.6.1.2">
<h4><span class="header-section-number">3.6.1.2</span> Dependencies</h4>
<p>We built co-assemblies of sequence data from plants in the same category; i.e., microbial sequences from all greenhouse-grown plants were co-assembled and those from all the field-grown were co-assembled.</p>
<p>KBase provides three assembly tools: <code>MetaSPAdes v3.15.3</code> <span class="citation">(<a href="#ref-Nurk2017" role="doc-biblioref">Nurk et al., 2017</a>)</span>, <code>MEGAHIT v1.2.9</code> <span class="citation">(<a href="#ref-Li2015" role="doc-biblioref">Li et al., 2015</a>)</span>, and <code>IDBA-UD v1.1.3</code> <span class="citation">(<a href="#ref-Peng2012" role="doc-biblioref">Peng et al., 2012</a>)</span>, and we used all three of these programs to assemble the metagenomic reads into contiguous sequences. See figure <a href="#fig:KbaseOverview">3.4</a></p>
<p>Assembling with multiple different tools is a common way to find which one produces the best results. Using <code>Compare Assembled Contig Distributions v1.1.2</code> software on Kbase, we saw that metaSPAdes has the best N50, N75, and it produced the longest contigs. See Figure <a href="#fig:KbaseOverview">3.4</a></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseCompareContigs"></span>
<img src="Images/Kbase_CompareContigs.png" alt="Result of Contig Comparison" width="100%" height="100%" />
<p class="caption">
Figure 3.8: Result of Contig Comparison
</p>
</div>
</div>
</div>
<div id="binning" class="section level3" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Binning</h3>
<div id="data-requirement-4" class="section level4" number="3.6.2.1">
<h4><span class="header-section-number">3.6.2.1</span> Data Requirement</h4>
<ul>
<li><p><strong>Input</strong>: PairedEndLibrary that was generated using the Import Application.</p></li>
<li><p><strong>Output</strong>: BinnedContig in <code>.fasta</code> format.</p></li>
</ul>
</div>
<div id="dependencies-5" class="section level4" number="3.6.2.2">
<h4><span class="header-section-number">3.6.2.2</span> Dependencies</h4>
<p>Just like with the assembly process, we used three binning programs: <code>MaxBin2 v2.2.4</code> <span class="citation">(<a href="#ref-Wu2016" role="doc-biblioref">Wu et al., 2015</a>)</span>, <code>MetaBAT2 v1.7</code> <span class="citation">(<a href="#ref-Kang2019" role="doc-biblioref">Kang et al., 2019</a>)</span>, and <code>CONCOCT v1.1</code> <span class="citation">(<a href="#ref-Alneberg2014" role="doc-biblioref">Alneberg et al., 2014</a>)</span> for MAG reconstruction. The individual outputs (bins) from these programs were passed into <code>DASTools- v1.1.2</code> to produce high-quality non-redundant MAGs. See figure <a href="#fig:KbaseOverview">3.4</a>.</p>
</div>
</div>
</div>
<div id="MAG_Classification" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> Classification of Metagenome Assembled Genomes</h2>
<div id="data-requirement-5" class="section level4" number="3.7.0.1">
<h4><span class="header-section-number">3.7.0.1</span> Data Requirement</h4>
<ul>
<li><p><strong>Input</strong>: BinnedContig in <code>.fasta</code> format</p></li>
<li><p><strong>Output</strong>: CheckM quality table in <code>.csv</code> format. Classification plot generated with SpeciesTree.</p></li>
</ul>
</div>
<div id="dependencies-6" class="section level4" number="3.7.0.2">
<h4><span class="header-section-number">3.7.0.2</span> Dependencies</h4>
<p>Prior to classification, <code>CheckM v1.0.18</code> was used to assess the quality of the MAGs and then removed the <code>Bin.003</code> with <strong>&gt;10% contamination</strong>. See table <a href="#tab:KbaseCheckM">3.2</a></p>
<table class=" lightable-paper lightable-hover" style="font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:KbaseCheckM">Table 3.2: </span>CheckM quality assessment of metagenome-assembled genomes that were recovered from greenhouse plants.
</caption>
<thead>
<tr>
<th style="text-align:center;">
Bin_Name
</th>
<th style="text-align:center;">
Marker_Lineage
</th>
<th style="text-align:center;">
Genomes_Size_.bp.
</th>
<th style="text-align:center;">
No_of_Markers
</th>
<th style="text-align:center;">
No_of_Marker_Sets
</th>
<th style="text-align:center;">
Completeness
</th>
<th style="text-align:center;">
Contamination
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
Bin.001
</td>
<td style="text-align:center;">
o__Sphingomonadales
</td>
<td style="text-align:center;">
3972965
</td>
<td style="text-align:center;">
569
</td>
<td style="text-align:center;">
293
</td>
<td style="text-align:center;">
98.62
</td>
<td style="text-align:center;">
0.90
</td>
</tr>
<tr>
<td style="text-align:center;">
Bin.002
</td>
<td style="text-align:center;">
c__Alphaproteobacteria
</td>
<td style="text-align:center;">
5311195
</td>
<td style="text-align:center;">
349
</td>
<td style="text-align:center;">
230
</td>
<td style="text-align:center;">
100.00
</td>
<td style="text-align:center;">
0.87
</td>
</tr>
<tr>
<td style="text-align:center;background-color: firebrick !important;">
Bin.003
</td>
<td style="text-align:center;background-color: firebrick !important;">
k__Archaea
</td>
<td style="text-align:center;background-color: firebrick !important;">
21065390
</td>
<td style="text-align:center;background-color: firebrick !important;">
149
</td>
<td style="text-align:center;background-color: firebrick !important;">
107
</td>
<td style="text-align:center;background-color: firebrick !important;">
70.02
</td>
<td style="text-align:center;background-color: firebrick !important;">
23.30
</td>
</tr>
</tbody>
<tfoot>
<tr>
<td style="padding: 0; border:0;" colspan="100%">
<sup>a</sup> The red highlighted area shows the low-quality bin that was filtered with the set parameter of &gt;50% completion and &lt;10% contamination
</td>
</tr>
</tfoot>
</table>
<p>Genome Taxonomy Database tool kit, <code>GTDB-Tk v1.7.0</code> <span class="citation">(<a href="#ref-Chaumeil2019" role="doc-biblioref">Chaumeil et al., 2019</a>)</span> was used to classify the other two bins.. See figure <a href="#fig:KbaseGTDB">3.9</a></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseGTDB"></span>
<img src="Images/Methagenome_Assembled_Genomes_from_GreenHouse_Plants.png" alt="Classification Tree generated by SpeciesTree - v2.2.0 on KBase" width="100%" height="100%" />
<p class="caption">
Figure 3.9: Classification Tree generated by SpeciesTree - v2.2.0 on KBase
</p>
</div>
<p>In addition to GTDB classification, we exported the fasta files of these bins from Kbase and classified them using Kraken2 and DIAMOND. We found that the Kraken2 and DIAMOND results were consistent with the GDTB-generated taxonomy but we will not go into the details of these software.</p>
</div>
</div>
<div id="FunctionalAnnotation" class="section level2" number="3.8">
<h2><span class="header-section-number">3.8</span> Gene Prediction and Functional profiling</h2>
<p>Gene prediction helps to predict the genetic potential of the recovered metagenome-assembled genomes.</p>
<div id="dram-annotation-on-kbase." class="section level3" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> DRAM Annotation on KBase.</h3>
<p>We used the <code>Annotate and Distill Assemblies with DRAM</code> software on KBase and the output can be seen in Figure <a href="#fig:KbaseDRAM">3.10</a></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:KbaseDRAM"></span>
<img src="Images/KBase.DRAM.Annotation.png" alt="DRAM Annotation of MAGS" width="100%" height="100%" />
<p class="caption">
Figure 3.10: DRAM Annotation of MAGS
</p>
</div>
</div>
<div id="gene-prediction" class="section level3" number="3.8.2">
<h3><span class="header-section-number">3.8.2</span> Gene Prediction</h3>
<p>We also annotated the MAGS using <code>Prodigal</code> <span class="citation">(<a href="#ref-Hyatt2010" role="doc-biblioref">Hyatt et al., 2010</a>)</span>, the annotated genomes were then passed into <code>KofamScan</code> (<a href="https://github.com/takaram/kofam_scan" class="uri">https://github.com/takaram/kofam_scan</a>), a command line interface of the web-based <code>KofamKOALA</code> <span class="citation">(<a href="#ref-Aramaki2019" role="doc-biblioref">Aramaki et al., 2019</a>)</span>, a gene function annotation tool based on KEGG Orthology and hidden Markov model. <code>MinPath</code> <span class="citation">(<a href="#ref-Ye2009" role="doc-biblioref">Ye and Doak, 2009</a>)</span> was used to reconstruct a parsimonious Kyoto Encyclopedia of Genes and Genomes, KEGG <span class="citation">(<a href="#ref-Kanehisa2000" role="doc-biblioref">Kanehisa and Goto, 2000</a>)</span> biological pathway using protein family predictions, this provides comprehensive reports on the metabolic modules and pathways that are found in MAGs.</p>
<div id="prodigal" class="section level4" number="3.8.2.1">
<h4><span class="header-section-number">3.8.2.1</span> Prodigal</h4>
<div id="data-requirement-6" class="section level5" number="3.8.2.1.1">
<h5><span class="header-section-number">3.8.2.1.1</span> Data Requirement</h5>
<ul>
<li><p><strong>Input</strong>: Genomes of the recovered bins in FASTA format. It can either be in <code>.fa</code> or <code>.fasta</code>.</p></li>
<li><p><strong>Output</strong>: <code>.fna</code>, <code>.ggf</code> and <code>.protein.faa</code> files. The protein file <code>.protein.faa</code> will be used as input for KOFAM Scan</p></li>
</ul>
</div>
<div id="dependencies-7" class="section level5" number="3.8.2.1.2">
<h5><span class="header-section-number">3.8.2.1.2</span> Dependencies</h5>
<p>If installed the <code>metagenomics</code> environment described in this <a href="#InstallMetagenomicsEnv">section</a>, just go ahead and activate the environment.</p>
<pre class="r"><code>conda activate metagenomics</code></pre>
<p>If you did not install that environment and you don’t want to, you can install prodigal directly.</p>
<pre class="r"><code>conda install -c bioconda prodigal</code></pre>
</div>
<div id="code-1" class="section level5" number="3.8.2.1.3">
<h5><span class="header-section-number">3.8.2.1.3</span> Code</h5>
<pre class="r"><code># Start a new tmux session

tmux new -s Functional.Annotation

# Activate the environment

conda activate metagenomics


# Make a new folder for the functional annotation

mkdir /path/to/Project_folder/06_Func_annotation

cd /path/to/Project_folder/06_Func_annotation


# Make a folder to store the FASTA files

mkdir /path/to/Project_folder/06_Func_annotation/01_bins.fasta

# Download the bins from KBase and move them into /path/to/Project_folder/06_Func_annotation/01_bins.fasta using FileZilla



# Run Prodigal on the bins 
# Run the program in normal mode. See https://github.com/hyattpd/prodigal/wiki/Advice-by-Input-Type#draft-genomes for more details.



# Create a for loop that runs the analysis for all of the bins  


for F in /path/to/Project_folder/06_Func_annotation/01_bins.fasta/*.fa; do
SAMPLE_NAME=$(basename &quot;${F}&quot; .fa)

mkdir /path/to/Project_folder/06_Func_annotation/${SAMPLE_NAME}_output

OUTPUT_DIR=/path/to/Project_folder/06_Func_annotation/${SAMPLE_NAME}_output

prodigal -a $OUTPUT_DIR/${SAMPLE_NAME}.protein.faa -d $OUTPUT_DIR/${SAMPLE_NAME}.contigs.fna -i $F  -o $OUTPUT_DIR/${SAMPLE_NAME}.gff

done</code></pre>
</div>
</div>
<div id="kofam-scan" class="section level4" number="3.8.2.2">
<h4><span class="header-section-number">3.8.2.2</span> KOFAM Scan</h4>
<div id="data-requirement-7" class="section level5" number="3.8.2.2.1">
<h5><span class="header-section-number">3.8.2.2.1</span> Data Requirement</h5>
<ul>
<li><p><strong>Input</strong>: Protein file <code>.faa</code> from Prodigal</p></li>
<li><p><strong>Output</strong>: KEGG Mapper file in <code>.txt</code>.</p></li>
</ul>
</div>
<div id="dependencies-8" class="section level5" number="3.8.2.2.2">
<h5><span class="header-section-number">3.8.2.2.2</span> Dependencies</h5>
<pre class="r"><code>## Run KOFAM

#Ref: https://taylorreiter.github.io/2019-05-11-kofamscan/

# Make a directory at /path/to/Project_folder/06_Func_annotation

mkdir /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan

cd /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan

wget ftp://ftp.genome.jp/pub/db/kofam/ko_list.gz        # download the ko list 
wget ftp://ftp.genome.jp/pub/db/kofam/profiles.tar.gz       # download the hmm profiles
wget ftp://ftp.genome.jp/pub/tools/kofam_scan/kofam_scan-1.3.0.tar.gz # download kofamscan tool
wget ftp://ftp.genome.jp/pub/tools/kofamscan/README.md      # download README


# unzip and untar the relevant files:

gunzip ko_list.gz

tar xf profiles.tar.gz

tar xf kofam_scan-1.3.0.tar.gz

## Make a directory that will store the outputs

mkdir /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/01_output

# Make a conda environment using miniconda

mamba create -n kofamscan

# activate the environment

mamba activate kofamscan

# Now, install ruby into the new environment

mamba install -c conda-forge ruby

# Also install kofamscan hmmer and parallel

mamba install kofamscan hmmer parallel




#make a config.yml file that has the directories, paste below and update with your own directories

vim config.yml

####PASTE BELOW INTO NEW YML AND UPDATE WITH NEW PATHS
# Path to your KO-HMM database
# A database can be a .hmm file, a .hal file or a directory in which
# .hmm files are. Omit the extension if it is .hal or .hmm file
profile: /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/profiles

# Path to the KO list file
ko_list: /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/ko_list

# Path to an executable file of hmmsearch
# You do not have to set this if it is in your $PATH
hmmsearch: ~/miniconda3/envs/kofamscan/bin/hmmsearch

# Path to an executable file of GNU parallel
# You do not have to set this if it is in your $PATH
parallel: ~/miniconda3/envs/kofamscan/bin/parallel

# Number of hmmsearch processes to be run parallelly
cpu: 48
####</code></pre>
</div>
<div id="code-2" class="section level5" number="3.8.2.2.3">
<h5><span class="header-section-number">3.8.2.2.3</span> Code</h5>
<pre class="r"><code># Use a for loop

for F in /path/to/Project_folder/06_Func_annotation/*Bin*/*.faa; do

SAMPLE_NAME=$(basename &quot;${F}&quot; .faa)

OUTPUT_DIR=/path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/01_output

~/miniconda3/pkgs/kofamscan-1.3.0-hdfd78af_2/bin/exec_annotation -c /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/config.yml $F -o $OUTPUT_DIR/${SAMPLE_NAME}.txt

done</code></pre>
</div>
</div>
<div id="minpath" class="section level4" number="3.8.2.3">
<h4><span class="header-section-number">3.8.2.3</span> MinPath</h4>
<div id="data-requirement-8" class="section level5" number="3.8.2.3.1">
<h5><span class="header-section-number">3.8.2.3.1</span> Data Requirement</h5>
<ul>
<li><p><strong>Input</strong>: KEGG Mapper file from KofamScan</p></li>
<li><p><strong>Output</strong>: KEGG Output file with annotated genes.</p></li>
</ul>
</div>
<div id="dependencies-9" class="section level5" number="3.8.2.3.2">
<h5><span class="header-section-number">3.8.2.3.2</span> Dependencies</h5>
<pre class="r"><code># Change to the folder

cd /path/to/Project_folder/06_Func_annotation

# Make a new folder for the MinPath annotation

mkdir 04_MinPath

cd 04_MinPath


## Copy the git repository that contain Minpath executable (https://github.com/mgtools/MinPath)

git clone https://github.com/mgtools/MinPath


# Use the information on this webpage to install glpk (https://github.com/mgtools/MinPath/blob/master/glpk-4.6/INSTALL)

# Move into the glpk folder

cd /path/to/Project_folder/06_Func_annotation/04_MinPath/MinPath/glpk-4.6

# Configuring the package

./configure


# Compile the package

make

# Check the package

make check

# Install the package. Note: By default, &#39;make install&#39; will install the package&#39;s files in
&#39;usr/local/bin&#39;, &#39;usr/local/lib&#39;. So, specify an installation
prefix other than &#39;/usr/local&#39;

make prefix=~/miniconda3/bin install

### make sure that you can call command glpsol under glpk-4.6/examples

/path/to/Project_folder/06_Func_annotation/04_MinPath/MinPath/glpk-4.6/examples

./glpsol -h</code></pre>
</div>
<div id="code-3" class="section level5" number="3.8.2.3.3">
<h5><span class="header-section-number">3.8.2.3.3</span> Code</h5>
<pre class="r"><code>## Now run the analysis
# Ref: https://github.com/mgtools/MinPath/blob/master/readme

# Make a folder for the input and output files

mkdir /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input

mkdir /path/to/Project_folder/06_Func_annotation/04_MinPath/02_output


# Copy the .txt files in &quot;/path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/01_output&quot; into &quot;/path/to/Project_folder/06_Func_annotation/04_MinPath/01_input&quot;


cp /path/to/Project_folder/06_Func_annotation/02_KOFAM_Scan/01_output/*txt /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input

# Change to /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input


cd /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input</code></pre>
<p>Format the KEGG Mapper files into a format that can be used by the MinPath program</p>
<pre class="r"><code>for F in /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input/*.txt; do

SAMPLE_NAME=$(basename &quot;${F}&quot; .txt)

OUTPUT_DIR=/path/to/Project_folder/06_Func_annotation/04_MinPath/01_input

##### convert this output to a form that can be used with KEGG decoder

awk &#39;{print $1, $2}&#39; $F &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper.txt


# Delete the first two rows of the data using tail command

tail -n +3 $OUTPUT_DIR/${SAMPLE_NAME}.mapper.txt &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper2.txt

# There are some rows that contains asterisks &quot;* &quot; in the first column, I will delete these strings but the rows using the awk command

awk &#39;$1 != &quot;*&quot; { print }&#39; $OUTPUT_DIR/${SAMPLE_NAME}.mapper2.txt &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper3.txt


## Turn the file into a tab-separated file

# Use sed or awk

sed &#39;s/ /\t/g&#39; $OUTPUT_DIR/${SAMPLE_NAME}.mapper3.txt &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper4.txt


awk &#39;BEGIN {FS=&quot; &quot;; OFS=&quot;\t&quot;} {$1=$1}1&#39; $OUTPUT_DIR/${SAMPLE_NAME}.mapper3.txt &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper4.txt


# In the second column, replace the rows that contains hypen &quot;-&quot; with nothing &quot;&quot; but do not delete the rows

awk &#39;{sub(/-/, &quot;&quot;, $2)}1&#39; $OUTPUT_DIR/${SAMPLE_NAME}.mapper4.txt &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapper5.txt

done</code></pre>
<p>Alternatively, you can make the code more concise by using the pipe (|) function for the codes above.</p>
<pre class="r"><code>for F in /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input/*.txt; do

SAMPLE_NAME=$(basename &quot;${F}&quot; .txt)

OUTPUT_DIR=/path/to/Project_folder/06_Func_annotation/04_MinPath/01_input

awk &#39;{print $1, $2}&#39; $F | tail -n +3 | awk &#39;$1 != &quot;*&quot; { print }&#39; | awk &#39;BEGIN {FS=&quot; &quot;; OFS=&quot;\t&quot;} {$1=$1}1&#39; | awk &#39;{sub(/-/, &quot;&quot;, $2)}1&#39; &gt; $OUTPUT_DIR/${SAMPLE_NAME}.mapperfile.txt

done</code></pre>
<p>Now that the Mapper files have been formated appropriately, let’s run the program. Check this <a href="https://github.com/mgtools/MinPath/blob/master/readme">link</a> for more details.</p>
<pre class="r"><code>for F in /path/to/Project_folder/06_Func_annotation/04_MinPath/01_input/*.mapperfile.txt; do

SAMPLE_NAME=$(basename &quot;${F}&quot; .txt)

OUTPUT_DIR=/path/to/Project_folder/06_Func_annotation/04_MinPath/02_output

python /path/to/Project_folder/06_Func_annotation/04_MinPath/MinPath/MinPath.py \
    -ko $F \
    -report $OUTPUT_DIR/${SAMPLE_NAME}.minpath \
    -details $OUTPUT_DIR/${SAMPLE_NAME}.minpath.details

done</code></pre>
<p>You will get a KEGG Output file with annotated genes that you can import into other programs.</p>
</div>
</div>
</div>
</div>
</div>
<div id="references" class="section level1" number="4">
<h1><span class="header-section-number">4</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Alneberg2014" class="csl-entry">
<span class="smallcaps">Alneberg, J.</span>, <span class="smallcaps">B.S. Bjarnason</span>, <span class="smallcaps">I. de Bruijn</span>, <span class="smallcaps">M. Schirmer</span>, <span class="smallcaps">J. Quick</span>, <span class="smallcaps">U.Z. Ijaz</span>, <span class="smallcaps">L. Lahti</span>, et al. 2014. Binning metagenomic contigs by coverage and composition. <em>Nat Methods</em> 11: 1144–1146.
</div>
<div id="ref-Anderson_and_Schluter_2021" class="csl-entry">
<span class="smallcaps">Andersen, S.B.</span>, and <span class="smallcaps">J. Schluter</span>. 2021. A metagenomics approach to investigate microbiome sociobiology. <em>Proceedings of the National Academy of Sciences</em> 118: e2100934118. Available at: <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2100934118">https://www.pnas.org/doi/abs/10.1073/pnas.2100934118</a>.
</div>
<div id="ref-Aramaki2019" class="csl-entry">
<span class="smallcaps">Aramaki, T.</span>, <span class="smallcaps">R. Blanc-Mathieu</span>, <span class="smallcaps">H. Endo</span>, <span class="smallcaps">K. Ohkubo</span>, <span class="smallcaps">M. Kanehisa</span>, <span class="smallcaps">S. Goto</span>, and <span class="smallcaps">H. Ogata</span>. 2019. <span class="nocase">KofamKOALA: KEGG Ortholog assignment based on profile HMM and adaptive score threshold</span>. <em>Bioinformatics</em> 36: 2251–2252. Available at: <a href="https://doi.org/10.1093/bioinformatics/btz859">https://doi.org/10.1093/bioinformatics/btz859</a>.
</div>
<div id="ref-Arkin2018" class="csl-entry">
<span class="smallcaps">Arkin, A.P.</span>, <span class="smallcaps">R.W. Cottingham</span>, <span class="smallcaps">C.S. Henry</span>, <span class="smallcaps">N.L. Harris</span>, <span class="smallcaps">R.L. Stevens</span>, <span class="smallcaps">S. Maslov</span>, <span class="smallcaps">P. Dehal</span>, et al. 2018. KBase: The united states department of energy systems biology knowledgebase. <em>Nature Biotechnology</em> 36: 566–569. Available at: <a href="https://doi.org/10.1038/nbt.4163">https://doi.org/10.1038/nbt.4163</a>.
</div>
<div id="ref-Berg2020" class="csl-entry">
<span class="smallcaps">Berg, G.</span>, <span class="smallcaps">D. Rybakova</span>, <span class="smallcaps">D. Fischer</span>, <span class="smallcaps">T. Cernava</span>, <span class="smallcaps">M.-C.C. Vergès</span>, <span class="smallcaps">T. Charles</span>, <span class="smallcaps">X. Chen</span>, et al. 2020. Microbiome definition re-visited: Old concepts and new challenges. <em>Microbiome</em> 8: 103. Available at: <a href="https://doi.org/10.1186/s40168-020-00875-0">https://doi.org/10.1186/s40168-020-00875-0</a>.
</div>
<div id="ref-Bharti2019" class="csl-entry">
<span class="smallcaps">Bharti, R.</span>, and <span class="smallcaps">D.G. Grimm</span>. 2019. Current challenges and best-practice protocols for microbiome analysis. <em>Briefings in Bioinformatics</em> 22: 178–193. Available at: <a href="https://api.semanticscholar.org/CorpusID:209409553">https://api.semanticscholar.org/CorpusID:209409553</a>.
</div>
<div id="ref-Browne2016" class="csl-entry">
<span class="smallcaps">Browne, H.P.</span>, <span class="smallcaps">S.C. Forster</span>, <span class="smallcaps">B.O. Anonye</span>, <span class="smallcaps">N. Kumar</span>, <span class="smallcaps">B.A. Neville</span>, <span class="smallcaps">M.D. Stares</span>, <span class="smallcaps">D. Goulding</span>, and <span class="smallcaps">T.D. Lawley</span>. 2016. Culturing of <span>“unculturable”</span> human microbiota reveals novel taxa and extensive sporulation. <em>Nature</em> 533: 543–546. Available at: <a href="https://doi.org/10.1038/nature17645">https://doi.org/10.1038/nature17645</a>.
</div>
<div id="ref-Buchfink2015" class="csl-entry">
<span class="smallcaps">Buchfink, B.</span>, <span class="smallcaps">C. Xie</span>, and <span class="smallcaps">D.H. Huson</span>. 2015. Fast and sensitive protein alignment using DIAMOND. <em>Nature Methods</em> 12: 59–60. Available at: <a href="https://doi.org/10.1038/nmeth.3176">https://doi.org/10.1038/nmeth.3176</a>.
</div>
<div id="ref-Chaumeil2019" class="csl-entry">
<span class="smallcaps">Chaumeil, P.-A.</span>, <span class="smallcaps">A.J. Mussig</span>, <span class="smallcaps">P. Hugenholtz</span>, and <span class="smallcaps">D.H. Parks</span>. 2019. <span class="nocase">GTDB-Tk: a toolkit to classify genomes with the Genome Taxonomy Database</span>. <em>Bioinformatics</em> 36: 1925–1927. Available at: <a href="https://doi.org/10.1093/bioinformatics/btz848">https://doi.org/10.1093/bioinformatics/btz848</a>.
</div>
<div id="ref-Gulati_2020" class="csl-entry">
<span class="smallcaps">Gulati, M.</span>, and <span class="smallcaps">B. Plosky</span>. 2020. As the microbiome moves on toward mechanism. <em>Molecular Cell</em> 78: 567. Available at: <a href="https://www.sciencedirect.com/science/article/pii/S1097276520303075">https://www.sciencedirect.com/science/article/pii/S1097276520303075</a>.
</div>
<div id="ref-Hyatt2010" class="csl-entry">
<span class="smallcaps">Hyatt, D.</span>, <span class="smallcaps">G.-L. Chen</span>, <span class="smallcaps">P.F. LoCascio</span>, <span class="smallcaps">M.L. Land</span>, <span class="smallcaps">F.W. Larimer</span>, and <span class="smallcaps">L.J. Hauser</span>. 2010. Prodigal: Prokaryotic gene recognition and translation initiation site identification. <em>BMC Bioinformatics</em> 11: 119. Available at: <a href="https://doi.org/10.1186/1471-2105-11-119">https://doi.org/10.1186/1471-2105-11-119</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
<span class="smallcaps">Kanehisa, M.</span>, and <span class="smallcaps">S. Goto</span>. 2000. <span class="nocase">KEGG: Kyoto Encyclopedia of Genes and Genomes</span>. <em>Nucleic Acids Research</em> 28: 27–30. Available at: <a href="https://doi.org/10.1093/nar/28.1.27">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Kang2019" class="csl-entry">
<span class="smallcaps">Kang, D.D.</span>, <span class="smallcaps">F. Li</span>, <span class="smallcaps">E. Kirton</span>, <span class="smallcaps">A. Thomas</span>, <span class="smallcaps">R. Egan</span>, <span class="smallcaps">H. An</span>, and <span class="smallcaps">Z. Wang</span>. 2019. <span>MetaBAT</span> 2: An adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies. <em>PeerJ</em> 7: e7359.
</div>
<div id="ref-Li2015" class="csl-entry">
<span class="smallcaps">Li, D.</span>, <span class="smallcaps">C.-M. Liu</span>, <span class="smallcaps">R. Luo</span>, <span class="smallcaps">K. Sadakane</span>, and <span class="smallcaps">T.-W. Lam</span>. 2015. <span class="nocase">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</span>. <em>Bioinformatics</em> 31: 1674–1676. Available at: <a href="https://doi.org/10.1093/bioinformatics/btv033">https://doi.org/10.1093/bioinformatics/btv033</a>.
</div>
<div id="ref-Lin2016" class="csl-entry">
<span class="smallcaps">Lin, H.-H.</span>, and <span class="smallcaps">Y.-C. Liao</span>. 2016. Accurate binning of metagenomic contigs via automated clustering sequences using information of genomic signatures and marker genes. <em>Scientific Reports</em> 6: 24175. Available at: <a href="https://doi.org/10.1038/srep24175">https://doi.org/10.1038/srep24175</a>.
</div>
<div id="ref-Melton2022" class="csl-entry">
<span class="smallcaps">Melton, A.E.</span>, <span class="smallcaps">A.W. Child</span>, <span class="smallcaps">J. Beard Richard S</span>, <span class="smallcaps">C.D.C. Dumaguit</span>, <span class="smallcaps">J.S. Forbey</span>, <span class="smallcaps">M. Germino</span>, <span class="smallcaps">M.-A. de Graaff</span>, et al. 2022. <span class="nocase">A haploid pseudo-chromosome genome assembly for a keystone sagebrush species of western North American rangelands</span>. <em>G3 Genes|Genomes|Genetics</em> 12: jkac122. Available at: <a href="https://doi.org/10.1093/g3journal/jkac122">https://doi.org/10.1093/g3journal/jkac122</a>.
</div>
<div id="ref-Miller2011" class="csl-entry">
<span class="smallcaps">Miller, R.</span> 2010. Characteristics of sagebrush habitats and limitations to long-term conservation.
</div>
<div id="ref-Nayfach2019" class="csl-entry">
<span class="smallcaps">Nayfach, S.</span>, <span class="smallcaps">Z.J. Shi</span>, <span class="smallcaps">R. Seshadri</span>, <span class="smallcaps">K.S. Pollard</span>, and <span class="smallcaps">N.C. Kyrpides</span>. 2019. New insights from uncultivated genomes of the global human gut microbiome. <em>Nature</em> 568: 505–510. Available at: <a href="https://doi.org/10.1038/s41586-019-1058-x">https://doi.org/10.1038/s41586-019-1058-x</a>.
</div>
<div id="ref-Nurk2017" class="csl-entry">
<span class="smallcaps">Nurk, S.</span>, <span class="smallcaps">D. Meleshko</span>, <span class="smallcaps">A. Korobeynikov</span>, and <span class="smallcaps">P.A. Pevzner</span>. 2017. <span class="nocase">metaSPAdes</span>: A new versatile metagenomic assembler. <em>Genome Res</em> 27: 824–834.
</div>
<div id="ref-Peng2012" class="csl-entry">
<span class="smallcaps">Peng, Y.</span>, <span class="smallcaps">H.C.M. Leung</span>, <span class="smallcaps">S.M. Yiu</span>, and <span class="smallcaps">F.Y.L. Chin</span>. 2012. <span class="nocase">IDBA-UD: a de novo assembler for single-cell and metagenomic sequencing data with highly uneven depth</span>. <em>Bioinformatics</em> 28: 1420–1428. Available at: <a href="https://doi.org/10.1093/bioinformatics/bts174">https://doi.org/10.1093/bioinformatics/bts174</a>.
</div>
<div id="ref-Remington2021" class="csl-entry">
<span class="smallcaps">Remington, T.E.</span>, <span class="smallcaps">P.A. Deibert</span>, <span class="smallcaps">S.E. Hanser</span>, <span class="smallcaps">D.M. Davis</span>, <span class="smallcaps">L.A. Robb</span>, and <span class="smallcaps">J.L. Welty</span>. 2021. Sagebrush conservation strategy—challenges to sagebrush conservation. U. S. G. Survey [ed.],. Reston, VA. Available at: <a href="https://doi.org/10.3133/ofr20201125">https://doi.org/10.3133/ofr20201125</a>.
</div>
<div id="ref-Rigge2020" class="csl-entry">
<span class="smallcaps">Rigge, M.</span>, <span class="smallcaps">C. Homer</span>, <span class="smallcaps">L. Cleeves</span>, <span class="smallcaps">D.K. Meyer</span>, <span class="smallcaps">B. Bunde</span>, <span class="smallcaps">H. Shi</span>, <span class="smallcaps">G. Xian</span>, et al. 2020. Quantifying western u.s. Rangelands as fractional components with multi-resolution remote sensing and in situ data. <em>Remote Sensing</em> 12: Available at: <a href="https://www.mdpi.com/2072-4292/12/3/412">https://www.mdpi.com/2072-4292/12/3/412</a>.
</div>
<div id="ref-Shipley2006" class="csl-entry">
<span class="smallcaps">Shipley, L.A.</span>, <span class="smallcaps">T.B. Davila</span>, <span class="smallcaps">N.J. Thines</span>, and <span class="smallcaps">B.A. Elias</span>. 2006. Nutritional requirements and diet choices of the pygmy rabbit (brachylagus idahoensis): A sagebrush specialist. <em>Journal of Chemical Ecology</em> 32: 2455–2474. Available at: <a href="https://doi.org/10.1007/s10886-006-9156-2">https://doi.org/10.1007/s10886-006-9156-2</a>.
</div>
<div id="ref-Wang2019" class="csl-entry">
<span class="smallcaps">Wang, Z.</span>, <span class="smallcaps">Z. Wang</span>, <span class="smallcaps">Y.Y. Lu</span>, <span class="smallcaps">F. Sun</span>, and <span class="smallcaps">S. Zhu</span>. 2019. <span class="nocase">SolidBin: improving metagenome binning with semi-supervised normalized cut</span>. <em>Bioinformatics</em> 35: 4229–4238. Available at: <a href="https://doi.org/10.1093/bioinformatics/btz253">https://doi.org/10.1093/bioinformatics/btz253</a>.
</div>
<div id="ref-Wood2019" class="csl-entry">
<span class="smallcaps">Wood, D.E.</span>, <span class="smallcaps">J. Lu</span>, and <span class="smallcaps">B. Langmead</span>. 2019. Improved metagenomic analysis with kraken 2. <em>Genome Biology</em> 20: 257. Available at: <a href="https://doi.org/10.1186/s13059-019-1891-0">https://doi.org/10.1186/s13059-019-1891-0</a>.
</div>
<div id="ref-Wu2016" class="csl-entry">
<span class="smallcaps">Wu, Y.-W.</span>, <span class="smallcaps">B.A. Simmons</span>, and <span class="smallcaps">S.W. Singer</span>. 2015. <span class="nocase">MaxBin 2.0: an automated binning algorithm to recover genomes from multiple metagenomic datasets</span>. <em>Bioinformatics</em> 32: 605–607. Available at: <a href="https://doi.org/10.1093/bioinformatics/btv638">https://doi.org/10.1093/bioinformatics/btv638</a>.
</div>
<div id="ref-Ye2009" class="csl-entry">
<span class="smallcaps">Ye, Y.</span>, and <span class="smallcaps">T.G. Doak</span>. 2009. A parsimony approach to biological pathway reconstruction/inference for genomes and metagenomes. <em>PLoS Comput Biol</em> 5: e1000465.
</div>
<div id="ref-Yu2018" class="csl-entry">
<span class="smallcaps">Yu, G.</span>, <span class="smallcaps">Y. Jiang</span>, <span class="smallcaps">J. Wang</span>, <span class="smallcaps">H. Zhang</span>, and <span class="smallcaps">H. Luo</span>. 2018. <span class="nocase">BMC3C: binning metagenomic contigs using codon usage, sequence composition and read coverage</span>. <em>Bioinformatics</em> 34: 4172–4179. Available at: <a href="https://doi.org/10.1093/bioinformatics/bty519">https://doi.org/10.1093/bioinformatics/bty519</a>.
</div>
</div>
<!-- We included this code so that we tell the computer to create another environment after our references. This is important because markdown is designed to have references at the end of the text-->
</div>
<div id="appendix-appendices" class="section level1 unnumbered">
<h1>Appendices</h1>
<!-- This code "(APPENDIX) Appendices {-}" was included so that Appendix will not be included as part of the table of content.  -->
</div>
<div id="appendix-1" class="section level1" number="5">
<h1><span class="header-section-number">A</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.25. 2023.
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>.</p>
<p>[2] S. M. Bache and H. Wickham. <em>magrittr: A Forward-Pipe Operator for
R</em>. R package version 2.0.3. 2022. <a href="https://magrittr.tidyverse.org" class="uri">https://magrittr.tidyverse.org</a>.</p>
<p>[3] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021.
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>.</p>
<p>[4] J. Cheng, C. Sievert, B. Schloerke, et al. <em>htmltools: Tools for
HTML</em>. R package version 0.5.7. 2023.
<a href="https://github.com/rstudio/htmltools" class="uri">https://github.com/rstudio/htmltools</a>.</p>
<p>[5] R. Francois and D. Hernangómez. <em>bibtex: Bibtex Parser</em>. R package
version 0.5.1. 2023. <a href="https://github.com/ropensci/bibtex" class="uri">https://github.com/ropensci/bibtex</a>.</p>
<p>[6] C. Glur. <em>data.tree: General Purpose Hierarchical Data Structure</em>.
R package version 1.1.0. 2023. <a href="https://github.com/gluc/data.tree" class="uri">https://github.com/gluc/data.tree</a>.</p>
<p>[7] R. Iannone. <em>DiagrammeR: Graph/Network Visualization</em>. R package
version 1.0.10. 2023. <a href="https://github.com/rich-iannone/DiagrammeR" class="uri">https://github.com/rich-iannone/DiagrammeR</a>.</p>
<p>[8] Y. Qiu. <em>prettydoc: Creating Pretty Documents from R Markdown</em>. R
package version 0.4.1. 2021. <a href="https://github.com/yixuan/prettydoc" class="uri">https://github.com/yixuan/prettydoc</a>.</p>
<p>[9] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[10] K. Ren and K. Russell. <em>formattable: Create Formattable Data
Structures</em>. R package version 0.2.1. 2021.
<a href="https://renkun-ken.github.io/formattable/" class="uri">https://renkun-ken.github.io/formattable/</a>.</p>
<p>[11] H. Wickham, J. Bryan, M. Barrett, et al. <em>usethis: Automate
Package and Project Setup</em>. R package version 2.2.2. 2023.
<a href="https://usethis.r-lib.org" class="uri">https://usethis.r-lib.org</a>.</p>
<p>[12] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of
Data Manipulation</em>. R package version 1.1.3. 2023.
<a href="https://dplyr.tidyverse.org" class="uri">https://dplyr.tidyverse.org</a>.</p>
<p>[13] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.5. 2022.
<a href="https://devtools.r-lib.org/" class="uri">https://devtools.r-lib.org/</a>.</p>
<p>[14] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2016. ISBN:
978-1138700109. <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>.</p>
<p>[15] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.36. 2023.
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>.</p>
<p>[16] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[17] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</p>
<p>[18] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.45. 2023.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[19] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2018. ISBN:
9781138359338. <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>.</p>
<p>[20] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. Boca
Raton, Florida: Chapman and Hall/CRC, 2020. ISBN: 9780367563837.
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>.</p>
<p>[21] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021.
<a href="http://haozhu233.github.io/kableExtra/" class="uri">http://haozhu233.github.io/kableExtra/</a>.</p>
</div>
<div id="appendix-2" class="section level1" number="6">
<h1><span class="header-section-number">B</span> Appendix 2</h1>
<p>Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using <code>sessionInfo()</code>.</p>
<pre><code>## R version 4.2.1 (2022-06-23)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur ... 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] DiagrammeR_1.0.10    data.tree_1.1.0      devtools_2.4.5      
##  [4] usethis_2.2.2        bibtex_0.5.1         knitcitations_1.0.12
##  [7] htmltools_0.5.7      prettydoc_0.4.1      magrittr_2.0.3      
## [10] dplyr_1.1.3          kableExtra_1.3.4     formattable_0.2.1   
## [13] bookdown_0.36        rmarkdown_2.25       knitr_1.45          
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.7         tidyr_1.3.0        sass_0.4.8         pkgload_1.3.3     
##  [5] jsonlite_1.8.8     viridisLite_0.4.2  bslib_0.6.1        shiny_1.8.0       
##  [9] highr_0.10         yaml_2.3.8         remotes_2.4.2.1    sessioninfo_1.2.2 
## [13] pillar_1.9.0       backports_1.4.1    glue_1.7.0         digest_0.6.33     
## [17] RColorBrewer_1.1-3 promises_1.2.1     rvest_1.0.3        RefManageR_1.4.0  
## [21] colorspace_2.1-0   httpuv_1.6.12      plyr_1.8.9         pkgconfig_2.0.3   
## [25] purrr_1.0.2        xtable_1.8-4       scales_1.3.0       webshot_0.5.5     
## [29] processx_3.8.2     svglite_2.1.2      later_1.3.1        timechange_0.2.0  
## [33] tibble_3.2.1       generics_0.1.3     ellipsis_0.3.2     withr_2.5.2       
## [37] cachem_1.0.8       cli_3.6.2          crayon_1.5.2       mime_0.12         
## [41] memoise_2.0.1      evaluate_0.23      ps_1.7.5           fs_1.6.3          
## [45] fansi_1.0.5        xml2_1.3.5         pkgbuild_1.4.2     profvis_0.3.8     
## [49] tools_4.2.1        prettyunits_1.2.0  formatR_1.14       lifecycle_1.0.4   
## [53] stringr_1.5.1      V8_4.4.0           munsell_0.5.0      callr_3.7.3       
## [57] compiler_4.2.1     jquerylib_0.1.4    systemfonts_1.0.5  rlang_1.1.3       
## [61] rstudioapi_0.15.0  visNetwork_2.1.2   htmlwidgets_1.6.2  miniUI_0.1.1.1    
## [65] curl_5.1.0         R6_2.5.1           lubridate_1.9.3    fastmap_1.1.1     
## [69] utf8_1.2.4         DiagrammeRsvg_0.1  stringi_1.8.3      Rcpp_1.0.11       
## [73] vctrs_0.6.5        tidyselect_1.2.0   xfun_0.41          urlchecker_1.0.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5,h6",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
